### PR TITLE
Geometry and others: update calls to use sdf::Errors parameters

### PR DESCRIPTION
**Pull request by marcoag**
Signed-off-by: Marco A. Gutierrez <marco@openrobotics.org>

# 🎉 New feature

Work towards https://github.com/gazebosim/sdformat/issues/820.

Depends on: https://github.com/gazebosim/sdformat/pull/1141.

Depends on: https://github.com/gazebosim/sdformat/pull/1263 and its forward port.

## Summary
Adds missing `Errors` structure parameters in a few methods of the following classes:

- Box
- Capsule
- Collision
- Cylinder
- Ellipsoid
- Geometry
- Heightmap
- Mesh
- Plane
- Polyline
- Sphere
- Surface

Adding several classes updates on this PR due to inter dependencies.

## Test it
Using the classes above should report all errors through `sdf::Errors` if the `errors` parameter is used, it should print them otherwise.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by marcoag** [comment]
When I thought I was almost done adding the `ToElement(errors)` and `Load()` method tests for the classes in this PR I found out the `ToElement` method of the `Surface` class is missing some elements. I added a PR that fixes this [here](https://github.com/gazebosim/sdformat/pull/1263).

I guess we should keep this PR on hold until that one is merged and ported to `sdf13`.

**Review by azeey — APPROVED** [review]

**Review by azeey — APPROVED** [review]

**Comment by azeey** [comment]
The test failures on windows are being investigated in https://github.com/gazebosim/sdformat/issues/1276.

**Pull request merged**
